flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/BootstrapTools.java

```diff
@@ -642,6 +654,35 @@ public static String getDynamicProperties(Configuration baseConfig, Configuratio
 		return String.join(" ", newAddedConfigs);
 	}
 
+	/**
+	 * Escape all the dynamic property values.
+	 * For unix-like OS(Linux, MacOS, FREE_BSD, etc.), each value will be surrounded with single quotes. This works for
+	 * all chars except single quote itself. To escape the single quote, close the quoting before it, insert the escaped
+	 * single quote, and then re-open the quoting. For example, the value is foo'bar and the escaped value is
+	 * 'foo'\''bar'.
+	 * For windows OS, each value will be surrounded with double quotes. The double quote itself needs to be escaped with
+	 * back slash. Also the caret symbol need to be escaped with double carets since it is the window command uses it to
+	 * escape characters.
```

Comment:
Could you add a reference to where you found this specification?

Reply:
I found it here https://en.wikibooks.org/wiki/Windows_Batch_Scripting. And run the test on window7.